src/lib.rs

```diff
@@ -214,6 +214,21 @@ mod zerocopy {
     pub(crate) use crate::*;
 }
 
+#[cfg(not(any(
+    target_arch = "i686",
+    target_arch = "x86",
+    target_arch = "x86_64",
+    target_arch = "arm",
+    target_arch = "aarch64",
+)))]
+compile_error!(
+r#"Zerocopy has not been proven sound on this architecture.
+
+This doesn't mean that it is unsound - just that we haven't gotten around to
+proving it sound. If you need support for this architecture, let us know and
+we'll prioritize it: https://github.com/google/zerocopy/issues/383"#
+);
```

Comment:
Re. this:
> TODO: Figure out how to make sure we block the next minor version release on adding back all of the targets that people actually rely on.

We could potentially take the initial, weaker step of merely warning on unproven architectures:
```suggestion
const _: () = {
    #[deprecated = r#"Zerocopy has not been proven sound on this architecture.

This doesn't mean that it is unsound - just that we haven't gotten around to
proving it sound. If you need support for this architecture, let us know and
we'll prioritize it: https://github.com/google/zerocopy/issues/383"#]
    const _WARNING: () = ();
    #[warn(deprecated)]
    _WARNING
};
```

Reply:
I think there's almost certainly merit in providing an escape hatch for weird architectures, too:
```suggestion
#[cfg(not(ZEROCOPY_ALLOW_EXPERIMENTAL_ARCH))]
compile_error!(
r#"Zerocopy has not been proven sound on this architecture.

This doesn't mean that it is unsound - just that we haven't gotten around to
proving it sound. If you need support for this architecture, let us know and
we'll prioritize it: https://github.com/google/zerocopy/issues/383

You can temporarily enable this architecture by setting:
export RUSTFLAGS="--cfg ZEROCOPY_ALLOW_EXPERIMENTAL_ARCH"
"#
);

#[cfg(ZEROCOPY_ALLOW_EXPERIMENTAL_ARCH)]
const _: () = {
    #[deprecated = r#"If you need support for this architecture, let us know and
we'll prioritize it: https://github.com/google/zerocopy/issues/383"#]
    const _WARNING: () = ();
    #[warn(deprecated)]
    _WARNING
};
```